services/svc_usb.py

```diff
@@ -91,21 +94,32 @@ def checkForInvalidKeywords(self):
           logging.debug("USB-Service: removing keyword '%s' because there are no images directly inside the basedir!" % keyword)
           self.removeKeywords(index)
       elif keyword not in self.getAllAlbumNames():
-        logging.info("USB-Service: removing invalid keyword: %s"%keyword)
+        logging.info("USB-Service: removing invalid keyword: %s" % keyword)
         self.removeKeywords(index)
       index -= 1
 
   def updateState(self):
-    if not os.path.exists(USB_Photos.BASE_DIR):
+    state = None
+    if not os.path.exists(self.baseDir):
       if not self.mountStorageDevice():
-        return BaseService.STATE_NOT_CONNECTED
-    if len(self.getAllAlbumNames()) == 0 and len(self.getBaseDirImages()) == 0:
-      self.unmountBaseDir()
-      return BaseService.STATE_NO_IMAGES
-    return BaseService.updateState(self)
+        state = BaseService.STATE_NOT_CONNECTED
```

Comment:
Should not use/create a new state in base service, needs to be stored locally. Ideally you simply use `STATE_NO_IMAGES` but have a local substate indicating why which you can use in `explainState()` below.

Reply:
i changed it